core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -0,0 +1,571 @@
+   * The Following is taken from ZIOFunctions.
+   * ------------------------------------------------------------------------------------------------
+   */
+  // ALL error types in this trait must be a subtype of `UpperE`.
+  // TAKEN FROM ZIO_E_THROWABLE
+  // ALL environment types in this trait must be a supertype of `LowerR`.
+  type IoR = Any
+
+  // type IoE = Polymorphic
+
+  /** Returns an effect that models failure with the specified error.
+   * The moral equivalent of `throw` for pure code.
+   */
+  final def fail[E](error: E): IO[E, Nothing] = ZIO.fail(error)
+
+  /** Returns an effect that models failure with the specified `Cause`.
+   */
+  final def halt[E](cause: Cause[E]): IO[E, Nothing] = ZIO.halt(cause)
+
+  /** Returns an effect that models success with the specified strictly-
+   * evaluated value.
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** Returns an effect that models success with the specified lazily-evaluated
+   * value. This method should not be used to capture effects. See
+   * `[[ZIO.effectTotal]]` for capturing total effects, and `[[ZIO.effect]]` for capturing
+   * partial effects.
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** Accesses the whole environment of the effect.
+   */
+  final def environment: ZIO[IoR, Nothing, IoR] = ZIO.access(ZIO.identityFn[IoR])
+
+  /** Accesses the environment of the effect.
+   * {{{
+   * val portNumber = effect.access(_.config.portNumber)
+   * }}}
+   */
+  final def access: ZIO.AccessPartiallyApplied[IoR] =
+    ZIO.access
+
+  /** Effectfully accesses the environment of the effect.
+   */
+  final def accessM: ZIO.AccessMPartiallyApplied[IoR] =
+    ZIO.accessM
+
+  /** Given an environment `R`, returns a function that can supply the
+   * environment to programs that require it, removing their need for any
+   * specific environment.
+   *
+   * This is similar to dependency injection, and the `provide` function can be
+   * thought of as `inject`.
+   */
+  final def provide[E, A](r: IoR): ZIO[IoR, E, A] => IO[E, A] =
+    ZIO.provide(r)
+
+  /** Returns an effect that accesses the runtime, which can be used to
+   * (unsafely) execute tasks. This is useful for integration with
+   * non-functional code that must call back into functional code.
+   */
+  final def runtime: ZIO[IoR, Nothing, Runtime[IoR]] = ZIO.runtime
+
+  /** Returns an effect that is interrupted.
+   */
+  final val interrupt: UIO[Nothing] = ZIO.interrupt
+
+  /** Returns a effect that will never produce anything. The moral
+   * equivalent of `while(true) {}`, only without the wasted CPU cycles.
+   */
+  final val never: UIO[Nothing] = ZIO.never
+
+  /** Returns an effect that dies with the specified `Throwable`.
+   * This method can be used for terminating a fiber because a defect has been
+   * detected in the code.
+   */
+  final def die(t: Throwable): UIO[Nothing] = ZIO.die(t)
```

Comment:
I'd delete all the comments in this file, later we can just link to ZIO documentation using the Scaladoc linking support:

```scala
/**
 * See [[scalaz.zio.ZIO.die]]
 */
```